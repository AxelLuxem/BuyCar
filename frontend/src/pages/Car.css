.carPageWrapper {
  /* background-color: red; */
  width: 100%;
  /* padding: 2rem 1rem; */
}

.topSection {
  display: flex;
  flex-wrap: wrap;
  justify-content: center;
  width: 100%;
}

.carImageBigWrapper {
  margin-right: 2.5rem;
  width: 50rem;
  @media (max-width: 900px) {
    width: 95%;
    margin-left: 0px;
    margin-right: 0px;
  }
}

.carImageBig {
  object-fit: contain;
  max-width: 800px;
  width: 100%;
  height: auto;
  border-radius: 10px;
}

.carDetailsWrapper {
  display: grid;
}

.carDetailsWrapper p {
  color: white;
}

.carDetailsName {
  font-size: 2.1875rem;
  font-weight: 600;
}

.carDetailsSpecs {
  font-size: 0.8125rem;
  margin-bottom: 2.5rem;
}

.priceText {
  font-size: 2.1875rem;
  font-weight: bold;
}

.contactSellerWrapper {
  background-color: #303030;
  padding: 0.625rem 2.1875rem;
  border-radius: 0.625rem;
  justify-self: flex-end;
}

.contactSellerText {
  font-size: 1.1875rem;
}

.contactSellerButton {
  display: flex;
  background-color: cadetblue;
  border-radius: 0.625rem;
  font-family: Verdana, Geneva, Tahoma, sans-serif;
  text-align: center;
  justify-content: center;
  align-items: center;
  padding: 0.9375rem 1.25rem;
  color: white;
}

.contactSellerButton:hover {
  background-color: #508688;
  cursor: pointer;
}
