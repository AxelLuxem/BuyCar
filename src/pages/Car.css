.carPageWrapper {
  /* background-color: red; */
  padding: 2rem 1rem;
}

.topSection {
  display: flex;
  flex-wrap: wrap;
  justify-content: center;
}

.carImageBigWrapper {
  margin: 0rem 2.5rem;
  width: 50rem;
  @media (max-width: 1100px) {
    width: 25rem;
    margin-left: 0px;
    margin-right: 0px;
  }
}

.carImageBig {
  object-fit: contain;
  max-width: 800px;
  width: 100%;
  height: auto;
}

.carDetailsWrapper {
  display: grid;
  flex-direction: column;
  flex-wrap: wrap;
}

.carDetailsWrapper p {
  color: white;
}

.carDetailsName {
  font-size: 2.1875rem;
  font-weight: 600;
}

.carDetailsSpecs {
  font-size: 0.8125rem;
  margin-bottom: 2.5rem;
}

.priceText {
  font-size: 2.1875rem;
  font-weight: bold;
}

.contactSellerWrapper {
  background-color: #303030;
  padding: 0.625rem 2.1875rem;
  border-radius: 0.625rem;
  justify-self: flex-end;
}

.contactSellerText {
  /* color: white; */
}

.contactSellerButton {
  display: flex;
  background-color: cadetblue;
  border-radius: 0.625rem;
  font-family: Verdana, Geneva, Tahoma, sans-serif;
  text-align: center;
  justify-content: center;
  align-items: center;
  padding: 0.9375rem 1.25rem;
  color: white;
}

.contactSellerButton:hover {
  background-color: #508688;
  cursor: pointer;
}
