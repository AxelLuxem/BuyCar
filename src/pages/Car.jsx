import { useEffect } from "react";
import { useParams } from "react-router-dom";
// import { carList } from "../carList";

const Car = () => {
  const params = useParams();

  return (
    <div>
      <h1>Welcome to the car page!!!</h1>
      <p>The id for this car is: {params.id}</p>
    </div>
  );
};

export default Car;
