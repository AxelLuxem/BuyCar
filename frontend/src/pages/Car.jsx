import { useParams } from "react-router-dom";
import { carList } from "../carList";
import "./Car.css";

const Car = () => {
  const carID = useParams();
  const path = "../pictures";

  const car = carList.find((item) => carID.id === item.id);

  return (
    <div className="carPageWrapper">
      <div className="topSection">
        <div className="carImageBigWrapper">
          <img
            src={`${path}/${car.pictureFilenames}`}
            alt={`${car.make} ${car.model}`}
            className="carImageBig"
            width="800px"
          />
        </div>
        <div className="carDetailsWrapper">
          <p className="carDetailsName">
            {car.make} {car.model}
          </p>
          <p className="carDetailsSpecs">
            {car.engineSize}litre {car.bhp}Bhp {car.doors}dr
          </p>
          <p>
            Year: {car.year} ({car.year[2] + car.year[3]}{" "}
            plate)
          </p>
          <p>Mileage: {car.miles} miles</p>
          <p>Previous owners: {car.ownerAmount}</p>
          <p className="priceText">Â£{car.price}</p>
          <div className="contactSellerWrapper">
            <p className="contactSellerText">
              {car.username}
            </p>
            <button className="contactSellerButton">
              Contact Seller
            </button>
          </div>
        </div>
      </div>
    </div>
  );
};

export default Car;
