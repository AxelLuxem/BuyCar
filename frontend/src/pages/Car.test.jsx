import { expect, describe, it, vi } from "vitest";
import { render } from "@testing-library/react";
import { useParams } from "react-router-dom";

import Car from "./Car";

vi.mock("react-router-dom");

describe("<Car>", () => {
  it("should render the component Car", () => {
    vi.mocked(useParams).mockReturnValue({ id: "04" });

    const result = render(<Car />);

    expect(result).toMatchSnapshot();
  });
});
