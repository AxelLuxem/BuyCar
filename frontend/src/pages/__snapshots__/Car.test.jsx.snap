// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<Car> > should render the component Car 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="carPageWrapper"
      >
        <div
          class="topSection"
        >
          <div
            class="carImageBigWrapper"
          >
            <img
              alt="Ford Mustang"
              class="carImageBig"
              src="../pictures/Car4.jpg"
              width="800px"
            />
          </div>
          <div
            class="carDetailsWrapper"
          >
            <p
              class="carDetailsName"
            >
              Ford
               
              Mustang
            </p>
            <p
              class="carDetailsSpecs"
            >
              5
              litre 
              296
              Bhp 
              2
              dr
            </p>
            <p>
              Year: 
              2004
               (
              04
               
              plate)
            </p>
            <p>
              Mileage: 
              30000
               miles
            </p>
            <p>
              Previous owners: 
              3
            </p>
            <p
              class="priceText"
            >
              £
              25000
            </p>
            <div
              class="contactSellerWrapper"
            >
              <p
                class="contactSellerText"
              >
                speedyRider
              </p>
              <button
                class="contactSellerButton"
              >
                Contact Seller
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="carPageWrapper"
    >
      <div
        class="topSection"
      >
        <div
          class="carImageBigWrapper"
        >
          <img
            alt="Ford Mustang"
            class="carImageBig"
            src="../pictures/Car4.jpg"
            width="800px"
          />
        </div>
        <div
          class="carDetailsWrapper"
        >
          <p
            class="carDetailsName"
          >
            Ford
             
            Mustang
          </p>
          <p
            class="carDetailsSpecs"
          >
            5
            litre 
            296
            Bhp 
            2
            dr
          </p>
          <p>
            Year: 
            2004
             (
            04
             
            plate)
          </p>
          <p>
            Mileage: 
            30000
             miles
          </p>
          <p>
            Previous owners: 
            3
          </p>
          <p
            class="priceText"
          >
            £
            25000
          </p>
          <div
            class="contactSellerWrapper"
          >
            <p
              class="contactSellerText"
            >
              speedyRider
            </p>
            <button
              class="contactSellerButton"
            >
              Contact Seller
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
